common/lib/protocol-definitions/src/storage.ts

```diff
@@ -117,6 +117,13 @@ export interface ITree {
 	 * Indicates that this tree is unreferenced. If this is not present, the tree is considered referenced.
 	 */
 	unreferenced?: true;
+
+	/**
+	 * Represents the group to which the tree belongs to. When the server returns the snapshot to
+	 * the client, and this tree is missing in that snapshot, then this could be used to request the contents
+	 * from the service.
+	 */
+	groupIdOrName?: string;
```

Comment:
groupIdOrName? Or here assumes there are two concepts (maybe merged into one), which is not the case.
I'd use either groupIdName or groupId.
Aso whatever is the name, let's make sure service uses the same name (though obviously we can remap as we do translation).